external-crates/move/crates/move-compiler/src/typing/macro_expand.rs

```diff
@@ -969,8 +979,12 @@ fn exp(context: &mut Context, sp!(eloc, e_): &mut N::Exp) {
         ///////
         N::Exp_::Var(sp!(_, v_)) if context.lambdas.contains_key(v_) => {
             context.mark_used(v_);
-            let (lambda, _, _) = context.lambdas.get(v_).unwrap();
```

Comment:
This case was the IDE bug. We were ignoring the types we had gathered for this lambda when threading it through between macro calls. This could be solved by eta-expanding lambdas in this case, but I figured to do something a bit different to handle annotations too